pyuvdata/parameter.py

```diff
@@ -225,6 +237,24 @@ def __eq__(self, other):
                     print(f"{self.name} parameter value is array, values are not close")
                     return False
             else:
+                # check to see if strict types are used
+                if self.strict_type:
+                    # types must match
+                    if not isinstance(self.value, other.expected_type):
+                        print(
+                            f"{self.name} parameter has incompatible types. Left is "
+                            f"{type(self.value)}, right is {other.expected_type}"
+                        )
+                        return False
+                if other.strict_type:
+                    # types must match in the other direction
+                    if not isinstance(other.value, self.expected_type):
+                        print(
+                            f"{self.name} parameter has incompatible types. Left is "
+                            f"{type(other.value)}, right is {other.expected_type}"
```

Comment:
same comment here about `other` being both left and right.

Reply:
Yes, I fixed this one and the one before to be more clear.